client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+
+interface IProps {
+  component: React.FC;
+  path: string;
+  exact: boolean;
+}
+
+const ProtectedRoute: React.FC<IProps> = props => {
+  const { loggedInUser } = useAuth();
+
+  return loggedInUser ? (
+    <Route path={props.path} exact={props.exact} component={props.component} />
```

Comment:
You don't need to pass these explicitly, just forwarding all the received props should suffice.